src/base_alloc/base_alloc_global.c

```diff
@@ -94,24 +91,16 @@ void *umf_ba_global_alloc(size_t size) {
         return ba_os_alloc(size);
     }
 
-    int ac_index = size_to_idx(size);
-    if (!BASE_ALLOC.ac[ac_index]) {
```

Comment:
I don't think you should remove this check. Creating ac can fail and there's no reason we can't serve the allocation using ba_os_alloc then. If you want to simplify the code, you can just combine this check with the one above.

Reply:
right, done